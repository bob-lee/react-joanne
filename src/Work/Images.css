.icons .icon {
  z-index: 214;
  position: fixed;
  top: calc(50% - 30px);
  height: 60px;
  width: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
  background: white;
  visibility: hidden;
  opacity: 0;
  transition: visibility .4s linear, opacity .4s linear, transform .3s linear;
}
.icons.show .icon {
  visibility: visible;
  opacity: 0.3;
}

.icons .icon.exit {
  top: 1em;
  left: 5px;
}
.icons .icon.prev {
  left: 5px;
}
.icons .icon.next {
  right: 5px;
}
.icons .icon:hover {
  cursor: pointer;
  transform:scale(1.05);
}
