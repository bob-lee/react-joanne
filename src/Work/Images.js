import React from 'react'
import Image from './Image'

const isWindow = typeof window !== 'undefined'
if (isWindow) {
  var scroll = require('react-scroll').animateScroll
}

class Images extends React.Component {
  state = { showIcon: false }
  lastY = 0
  ticking = false

  scrollToTop = () => scroll.scrollToTop()
  currentPositionY = () => window.pageYOffset

  updateLastY() {
    const newY = this.currentPositionY()
    if (newY !== this.lastY) {
      this.lastY = newY
      this.setState({
        showIcon: isWindow && this.lastY > 500
      })
      //console.log('updateLastY', Math.ceil(newY))
    }
  }

  handleScroll = (e) => {
    if(!this.ticking) {
      window.requestAnimationFrame(() => {
        this.updateLastY()
        this.ticking = false
      })
      this.ticking = true
    }
  }

  componentDidMount() {
    isWindow && window.addEventListener('scroll', this.handleScroll)
  }

  componentWillUnmount() {
    isWindow && window.removeEventListener('scroll', this.handleScroll)
  }

  render() {
    const { list, ...propsButList } = this.props
    const classes = `back-to-top ${this.state.showIcon ? 'show' : ''}`
    const info = Math.ceil(this.lastY)

    return (
      <div>
        <div className="images">
          {list.map((item, index) =>
            <Image key={item.fileName}
              item={item}
              index={index}
              {...propsButList} />
          )}
        </div>
        <div className={classes} onClick={this.scrollToTop}>
          <i className="fa fa-step-forward fa-lg" aria-hidden="true"></i>
        </div>
        <div className="back-to-top test">
          {info}
        </div>
      </div>
    )
  }
}

export default Images
