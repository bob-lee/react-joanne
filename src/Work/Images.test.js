import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import Images from './Images';

export const list = [
  {
    fileName: 'test1.jpg',
    url: '/test1.jpg',
    thumbUrl: '/test1_thumb.jpg',
    text: 'test1',
    toLoad: true
  },
  {
    fileName: 'test2.jpg',
    url: '/test2.jpg',
    thumbUrl: '/test2_thumb.jpg',
    text: 'test2',
    toLoad: false
  }
]

const handleImageLoaded = jest.fn()

it('Images renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Images list={list} onImageLoaded={handleImageLoaded} />, div);
});

it('Images renders correctly', () => {
  const tree = renderer
    .create(<Images list={list} onImageLoaded={handleImageLoaded} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// it(`Images renders correctly`, () => {
//   const wrapper = shallow(<Images list={list} onImageLoaded={handleImageLoaded} />);
//   expect(toJson(wrapper)).toMatchSnapshot();
// });

it(`Images renders correctly after first image loaded`, () => {
  const wrapper = mount(<Images list={list} onImageLoaded={handleImageLoaded} />);
  const firstChild = wrapper.find('.images').childAt(0)
  //console.log(firstChild)
  firstChild.find('img.image1').simulate('load');
  expect(toJson(wrapper)).toMatchSnapshot();
});
