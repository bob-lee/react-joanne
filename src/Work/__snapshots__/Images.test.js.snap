// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images renders correctly 1`] = `
<div>
  <div
    className="images"
  >
    <div
      data-index={0}
      name="test1.jpg"
    >
      <div>
        <div
          className="loader"
          style={
            Object {
              "margin": "0 auto",
            }
          }
        >
          Loading...
        </div>
        <div
          className="image"
        >
          <a
            aria-hidden="true"
            className="anchor"
            href="#test1.jpg"
            onClick={[Function]}
          >
            <picture>
              <img
                alt="test1.jpg"
                className="image1"
                onLoad={[Function]}
                src="/test1.jpg"
                title="image"
              />
            </picture>
          </a>
          <div
            className="expand"
          >
            <span>
              test1
            </span>
            <img
              alt="thumb"
              className="image2"
              src="/test1_thumb.jpg"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      data-index={1}
      name="test2.jpg"
    >
      <div>
        <div
          className="image"
        >
          <a
            aria-hidden="true"
            className="anchor"
            href="#test2.jpg"
            onClick={[Function]}
          >
            <picture>
              <img
                alt="test2.jpg"
                className="image1"
                onLoad={[Function]}
                title="image"
              />
            </picture>
          </a>
          <div
            className="expand"
          >
            <span>
              test2
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="back-to-top"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-step-forward fa-lg"
    />
  </div>
  <div
    className="back-to-top test"
  >
    0
  </div>
  <div
    className="icons"
  >
    <div
      className="icon exit"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-times fa-w-11"
        data-icon="times"
        data-prefix="fas"
        focusable="false"
        height="24"
        role="img"
        viewBox="0 0 352 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      className="icon prev"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-left fa-w-10"
        data-icon="chevron-left"
        data-prefix="fas"
        focusable="false"
        height="24"
        role="img"
        viewBox="0 0 320 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      className="icon next"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-right fa-w-10"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        height="24"
        role="img"
        viewBox="0 0 320 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    className="fullscreen1"
    id="fullscreen"
  >
    <canvas
      id="canvas"
      onClick={[Function]}
    />
  </div>
  <div
    className="fullscreen2"
    id="fullscreen"
  >
    <canvas
      id="canvas"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Images renders correctly after first image loaded 1`] = `
<Images
  hash=""
  list={
    Array [
      Object {
        "fileName": "test1.jpg",
        "text": "test1",
        "thumbUrl": "/test1_thumb.jpg",
        "toLoad": true,
        "url": "/test1.jpg",
      },
      Object {
        "fileName": "test2.jpg",
        "text": "test2",
        "thumbUrl": "/test2_thumb.jpg",
        "toLoad": false,
        "url": "/test2.jpg",
      },
    ]
  }
  workOnClick={[MockFunction]}
  workOnLoad={
    [MockFunction] {
      "calls": Array [
        Array [
          "load",
          Object {
            "element": <img
              alt="test1.jpg"
              class="image1"
              src="/test1.jpg"
              title="image"
            />,
            "index": 0,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div>
    <div
      className="images"
    >
      <Element
        data-index={0}
        key="test1.jpg"
        name="test1.jpg"
      >
        <ElementWrapper
          data-index={0}
          name="test1.jpg"
          parentBindings={
            Object {
              "domNode": <div
                data-index="0"
                name="test1.jpg"
              >
                <div>
                  <div
                    class="image show"
                  >
                    <a
                      aria-hidden="true"
                      class="anchor"
                      href="#test1.jpg"
                    >
                      <picture>
                        <img
                          alt="test1.jpg"
                          class="image1"
                          src="/test1.jpg"
                          title="image"
                        />
                      </picture>
                    </a>
                    <div
                      class="expand"
                    >
                      <span>
                        test1
                      </span>
                      <img
                        alt="thumb"
                        class="image2"
                        src="/test1_thumb.jpg"
                      />
                    </div>
                  </div>
                </div>
              </div>,
            }
          }
        >
          <div
            data-index={0}
            name="test1.jpg"
          >
            <Image
              imagesOnClick={[Function]}
              imagesOnLoad={[Function]}
              index={0}
              item={
                Object {
                  "fileName": "test1.jpg",
                  "text": "test1",
                  "thumbUrl": "/test1_thumb.jpg",
                  "toLoad": true,
                  "url": "/test1.jpg",
                }
              }
            >
              <div>
                <div
                  className="image show"
                >
                  <a
                    aria-hidden="true"
                    className="anchor"
                    href="#test1.jpg"
                    onClick={[Function]}
                  >
                    <picture>
                      <img
                        alt="test1.jpg"
                        className="image1"
                        onLoad={[Function]}
                        src="/test1.jpg"
                        title="image"
                      />
                    </picture>
                  </a>
                  <div
                    className="expand"
                  >
                    <span>
                      test1
                    </span>
                    <img
                      alt="thumb"
                      className="image2"
                      src="/test1_thumb.jpg"
                    />
                  </div>
                </div>
              </div>
            </Image>
          </div>
        </ElementWrapper>
      </Element>
      <Element
        data-index={1}
        key="test2.jpg"
        name="test2.jpg"
      >
        <ElementWrapper
          data-index={1}
          name="test2.jpg"
          parentBindings={
            Object {
              "domNode": <div
                data-index="1"
                name="test2.jpg"
              >
                <div>
                  <div
                    class="image"
                  >
                    <a
                      aria-hidden="true"
                      class="anchor"
                      href="#test2.jpg"
                    >
                      <picture>
                        <img
                          alt="test2.jpg"
                          class="image1"
                          title="image"
                        />
                      </picture>
                    </a>
                    <div
                      class="expand"
                    >
                      <span>
                        test2
                      </span>
                    </div>
                  </div>
                </div>
              </div>,
            }
          }
        >
          <div
            data-index={1}
            name="test2.jpg"
          >
            <Image
              imagesOnClick={[Function]}
              imagesOnLoad={[Function]}
              index={1}
              item={
                Object {
                  "fileName": "test2.jpg",
                  "text": "test2",
                  "thumbUrl": "/test2_thumb.jpg",
                  "toLoad": false,
                  "url": "/test2.jpg",
                }
              }
            >
              <div>
                <div
                  className="image"
                >
                  <a
                    aria-hidden="true"
                    className="anchor"
                    href="#test2.jpg"
                    onClick={[Function]}
                  >
                    <picture>
                      <img
                        alt="test2.jpg"
                        className="image1"
                        onLoad={[Function]}
                        title="image"
                      />
                    </picture>
                  </a>
                  <div
                    className="expand"
                  >
                    <span>
                      test2
                    </span>
                  </div>
                </div>
              </div>
            </Image>
          </div>
        </ElementWrapper>
      </Element>
    </div>
    <div
      className="back-to-top"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-step-forward fa-lg"
      />
    </div>
    <div
      className="back-to-top test"
    >
      0
    </div>
    <div
      className="icons"
    >
      <div
        className="icon exit"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-times fa-w-11"
          data-icon="times"
          data-prefix="fas"
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 352 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className="icon prev"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chevron-left fa-w-10"
          data-icon="chevron-left"
          data-prefix="fas"
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 320 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className="icon next"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chevron-right fa-w-10"
          data-icon="chevron-right"
          data-prefix="fas"
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 320 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <Fullscreen
      cls="fullscreen1"
      eId="fullscreen1"
      hideFullscreen={[Function]}
      image={null}
      nextFullscreen={[Function]}
      prevFullscreen={[Function]}
    >
      <div
        className="fullscreen1"
        id="fullscreen"
      >
        <canvas
          id="canvas"
          onClick={[Function]}
        />
      </div>
    </Fullscreen>
    <Fullscreen
      cls="fullscreen2"
      eId="fullscreen2"
      hideFullscreen={[Function]}
      image={null}
      nextFullscreen={[Function]}
      prevFullscreen={[Function]}
    >
      <div
        className="fullscreen2"
        id="fullscreen"
      >
        <canvas
          id="canvas"
          onClick={[Function]}
        />
      </div>
    </Fullscreen>
  </div>
</Images>
`;
